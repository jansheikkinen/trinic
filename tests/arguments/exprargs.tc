// Expression Arguments

function main(void) -> void do
  let x, y, z: int8 = undefined;
  let a, b, c: int8 = 1 + 2, 3, 4;
end
