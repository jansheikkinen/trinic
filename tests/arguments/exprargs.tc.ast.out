
### AST GENERATION ###
Declaration FUNCTION  main
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg x
      Identifier_Arg y
      Identifier_Arg z
        Type TYPE INT8
    Arg EXPR
      Expr UNDEFINED
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg a
      Identifier_Arg b
      Identifier_Arg c
        Type TYPE INT8
    Arg EXPR
      Expr BINARY +
        Expr LITERAL
          INT 1
        Expr LITERAL
          INT 2
      Expr LITERAL
        INT 3
      Expr LITERAL
        INT 4
